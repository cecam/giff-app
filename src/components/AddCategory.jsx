import {useState} from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({setCategories}) => {
    const [input, setInput] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault()
        
        if(input.trim().length > 2) {
            setCategories(categories => [input, ...categories])
            setInput('')
        }
    }

    return ( 
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                value={input}
                onChange={e => setInput(e.target.value)}
            />
        </form>
     );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
 
export default AddCategory;