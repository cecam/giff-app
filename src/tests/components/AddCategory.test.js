import React from 'react';
import { shallow } from 'enzyme'

import AddCategory from '../../components/AddCategory'

describe('Tests on <AddCategory />', () => {

    const setCategories = jest.fn() //way to declare a jest function
    let wrapper = shallow( <AddCategory setCategories={setCategories} /> )
    
    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow( <AddCategory setCategories={setCategories} /> )
    })

    test('should print it correctly', () => {

        expect( wrapper ).toMatchSnapshot()
    });

    test('should change the text input ', () => {
        const input = wrapper.find('input')
        const value = 'hello word'

        input.simulate('change', { target: {value} })

        expect( wrapper.find('input').prop('value') ).toBe(value)
    });

    test('should not post info on submit until input value be larger than 2', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled()

    });

    test('should post the info if the input value length is larger than 2 and clean the value after submit', () => {
        const value = 'hello word'
        
        wrapper.find('input').simulate('change', { target: {value} })

        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).toHaveBeenCalled()

        expect( wrapper.find('input').prop('value') ).toBe('')
    });
})