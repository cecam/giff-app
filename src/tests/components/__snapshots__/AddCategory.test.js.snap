// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> should print it correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
